src/Airship/Internal/Helpers.hs

```diff
@@ -22,12 +23,25 @@ import qualified Network.HTTP.Types        as HTTP
 import qualified Network.Wai               as Wai
 import           Network.Wai.Parse
 import           System.Random
+import           Text.Read
 
+import           Airship.Headers
 import           Airship.Internal.Decision
 import           Airship.Internal.Route
 import           Airship.Resource
 import           Airship.Types
-import           Airship.Headers
+
+-- | Reads the parameter of the provided name and attempt to convert
+-- it to the desired 'Read'-implementing type. If the parameter is not
+-- found or cannot be read successfully, the provided handler is invoked
+-- (this may often be `halt status404`).
```

Comment:
I would update the doc to suggest a different status code, likely HTTP 400. I don't think 404 makes sense for a missing/invalid query parameter, since the _resource_ _does_ exist.